pkg/bootstrap/platform/gcp.go

```diff
@@ -56,7 +56,7 @@ const (
 var GCPStaticMetadata = func() map[string]string {
 	gcpm := env.Register("GCP_METADATA", "", "Pipe separated GCP metadata, schemed as PROJECT_ID|PROJECT_NUMBER|CLUSTER_NAME|CLUSTER_ZONE").Get()
 	quota := env.Register("GCP_QUOTA_PROJECT", "", "Allows specification of a quota project to be used in requests to GCP APIs.").Get()
-	zone := env.Register("GCP_ZONE", "", "GCP Zone where the workload is running on.").Get()
+	zone := env.Register("GCP_ZONE", zoneFromResolvConf(), "GCP Zone where the workload is running on.").Get()
```

Comment:
by setting this as default it's going to run for all users - even non GCP users. Probably need to make it run in a more lazy context.

Reply:
This is not new, so could make all `GCPStaticMetadata` as a function

Reply:
I moved it to NewGCP function, so that only it runs in GCP.